Engine_Revit_UI/Convert/FromSI.cs

```diff
@@ -32,52 +32,10 @@ public static partial class Convert
-                case UnitType.UT_Length:
-                case UnitType.UT_Bar_Diameter:
-                case UnitType.UT_Section_Dimension:
-                case UnitType.UT_Section_Property:
-                case UnitType.UT_PipeSize:
-                case UnitType.UT_HVAC_DuctSize:
-                case UnitType.UT_HVAC_DuctLiningThickness:
-                case UnitType.UT_HVAC_DuctInsulationThickness:
-                case UnitType.UT_PipeInsulationThickness:
-                case UnitType.UT_SheetLength:
-                case UnitType.UT_WireSize:
-                case UnitType.UT_Crack_Width:
-                case UnitType.UT_DecSheetLength:
-                case UnitType.UT_Electrical_CableTraySize:
-                case UnitType.UT_Electrical_ConduitSize:
-                case UnitType.UT_Reinforcement_Length:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_METERS);
-                case UnitType.UT_Mass:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_KILOGRAMS_MASS);
-                case UnitType.UT_Electrical_Current:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_AMPERES);
-                case UnitType.UT_HVAC_Temperature:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_KELVIN);
-                case UnitType.UT_Weight:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_KILOGRAMS_MASS);
-                case UnitType.UT_HVAC_Pressure:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_PASCALS);
-                case UnitType.UT_Piping_Pressure:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_PASCALS);
-                case UnitType.UT_HVAC_Velocity:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_METERS_PER_SECOND);
-                case UnitType.UT_Area:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_SQUARE_METERS);
-                case UnitType.UT_Volume:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_CUBIC_METERS);
-                case UnitType.UT_HVAC_Airflow:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_CUBIC_METERS_PER_SECOND);
-                case UnitType.UT_HVAC_Cooling_Load:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_WATTS);
-                case UnitType.UT_HVAC_Heating_Load:
-                    return UnitUtils.ConvertToInternalUnits(Value, DisplayUnitType.DUT_WATTS);
-                default:
-                    return Value;
-            }
+            if (double.IsNaN(Value) || Value == double.MaxValue || Value == double.MinValue || double.IsNegativeInfinity(Value) || double.IsPositiveInfinity(Value))
+                return Value;
```

Comment:
Local variable should have a lowercase letter - `value` not `Value`

Reply:
Agreed, that is Jakub's legacy - decided not to touch it to avoid backward compatibility issues. Or is it safe to change method argument names?

Reply:
Should be safe, but good shout, let's leave it for this PR